### PR TITLE
storage: add tool for reflashing all MCUs

### PR DESCRIPTION
**Pull request by nest1ing**
This brings `yadro-mcu-reflash` that provides a way for reflashing MCU on all installed backplanes.

Typical usage is a call via `ExecStartPre` in `yadro-storage.service`. But as it requires the MCU firmware images and a special config file this is not specified as a default action and will be added by a separate recipe on yocto side.

Signed-off-by: Alexander Filippov <a.filippov@yadro.com>

### CODE REVIEW

File: src/mcu/reflash.cpp [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+                64; // image can't be less then 64 bytes (header size)
+            static constexpr size_t maxImageSize =
+                128 * 1024; // image can't be more then 128 Kbytes
+            if ((fw.size() < minImageSize) || (fw.size() > maxImageSize))
+            {
+                fprintf(stderr, "Incorrect '%s' size\n", firmware.c_str());
+                return;
+            }
+
+            mcu->eraseFlash();
```
Comment (AlexanderAmelkin) [inline src/mcu/reflash.cpp]: I thought you wanted to skip this

Comment (nest1ing) [reply src/mcu/reflash.cpp]: That happened later that this pull-request created. :)
Fixed.

File: src/common/mmapfile.hpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#pragma once
+
+#include <cstddef>
+#include <string>
+
+namespace common
+{
+
+/**
+ * @brief RAI wrapper for mmap()/munmap()
```
Comment (AlexanderAmelkin) [inline src/common/mmapfile.hpp]: RAII ?

File: src/mcu/reflash.cpp [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+    }
+    return -1;
+}
+
+static constexpr size_t numberOfPins = 8;
+
+/**
+ * @brief Calculate value encoded by bits
+ *
+ * NOTE: It uses the reverse format when 0-th bit is the lest.
```
Comment (AlexanderAmelkin) [inline src/mcu/reflash.cpp]: `is the leftmost`

**Review by AlexanderAmelkin — DISMISSED** [review]

File: src/common/mmapfile.hpp (hunk 2 above)
Comment (nest1ing) [reply src/common/mmapfile.hpp]: fixed

File: src/mcu/reflash.cpp (hunk 3 above)
Comment (nest1ing) [reply src/mcu/reflash.cpp]: fixed

**Review by AlexanderAmelkin — APPROVED** [review]

**Pull request merged**